### PR TITLE
Tidy up the Dockerfile

### PR DESCRIPTION
**Pull request by damacus**
- Uses `&&` to join commands as a standard.
- Install packages directly from the Dockerfile rather than copying in a file which we then delete. This makes it easier for the operator to understand what is being installed
- As a convention set multiple environment variable in one. Rather than switching
- Install all apt packages in one set to speed up installation

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -6,103 +6,131 @@ ARG NCPUS=1
...
+    alien \
+    build-essential \
+    bzip2 \
+    ca-certificates \
+    cargo \
+    curl \
+    nano \
+    vim \
+    wget \
+    git \
```
Comment (xoen) [inline Dockerfile:33]: I'd probably keep these in alphabetical order to make it easier to read/update. (not that Docker or `apt` cares ☺️ )

File: Dockerfile [hunk 2]
```diff
@@ -6,103 +6,131 @@ ARG NCPUS=1
...
-#    # Install phantomjs via webshot for Doc/PDF with JS graphs in it
     && R -e "library(webshot); webshot::install_phantomjs()" \
     && mv /root/bin/phantomjs /usr/bin/phantomjs \
     && chmod a+rx /usr/bin/phantomjs
 
 RUN npm config set unsafe-perm true \
     && npm install -g vega vega-lite
 
-# We want all packages to use the default MRAN mirror so that when we upgrade users, they don't magically get new packages
+# We want all packages to use the default MRAN mirror so that when we upgrade users, they dont magically get new packages
```
Comment (xoen) [inline Dockerfile:119]: interesting how the `don't` was changed to `dont` 😬

Comment (damacus) [reply Dockerfile:119]: :D what now, fixed

File: Makefile [hunk 3]
```diff
@@ -1,14 +1,14 @@
 SHELL = '/bin/bash'
-export BUILD_TAG ?= 1.2.1335-r3.5.1-python3.7.1-conda-10
+export BUILD_TAG ?= local
```
Comment (xoen) [inline Makefile:2]: nice, it makes sense as for example I've been changing this and the `docker-compose.yml` file locally ☺️ 

Speaking of `docker-compose.yml`, do you reckon the tag should match this as well there?

File: Dockerfile [hunk 4]
```diff
@@ -6,75 +6,109 @@ ARG NCPUS=1
...
-ENV LC_ALL=en_GB.UTF-8 \
-    LANG=en_GB.UTF-8
-
-ENV PATH /opt/conda/bin:$PATH
+ENV R_VERSION=3.5.1 \
+    PY_VERSION=${PY_VERSION:-3.7} \
+    USER=rstudio \
+    LC_ALL=en_GB.UTF-8 \
+    LANG=en_GB.UTF-8 \
+    PATH=/opt/conda/bin:$PATH
```
Comment (xoen) [inline Dockerfile:14]: I guess one benefit of having these set in separate `ENV` commands (and I guess these are relatively independent from each other) is that if we add an environment variable at the end of this list the previous layers would be cached.

But to be fair, `ENV` commands are pretty quick and these are also basically at the top of the `Dockerfile` so they would invalidate a lot of layers anyway 👌

Comment (damacus) [reply Dockerfile:14]: That was my exact thought process! We had a mix of methods so I just picked one and stuck with it.

File: Makefile (hunk 3 above)
Comment (damacus) [reply Makefile:2]: The docker-compose.yml tag will inherit from here, so I think we're good in this case

Comment (xoen) [reply Makefile:2]: Oh nice, [I didn't see this](https://github.com/ministryofjustice/analytics-platform-rstudio/blob/main/docker-compose.yml#L6), I assumed it was same as in the `jupyter-lab` image ❤️

**Pull request merged**